osu.Game/Overlays/FullscreenOverlay.cs

```diff
@@ -27,37 +29,58 @@ public abstract class FullscreenOverlay<T> : WaveOverlayContainer, INamedOverlay
         [Cached]
         protected readonly OverlayColourProvider ColourProvider;
 
+        [Cached]
+        protected OngoingOperationTracker OngoingOperationTracker { get; private set; }
+
+        private readonly LoadingLayer loadingLayer;
+
         protected FullscreenOverlay(OverlayColourScheme colourScheme, T header)
         {
             Header = header;
-
             ColourProvider = new OverlayColourProvider(colourScheme);
 
+            AddRangeInternal(new Drawable[]
+            {
+                loadingLayer = new LoadingLayer(true),
+                OngoingOperationTracker = new OngoingOperationTracker()
+            });
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
             RelativeSizeAxes = Axes.Both;
             RelativePositionAxes = Axes.Both;
             Width = 0.85f;
             Anchor = Anchor.TopCentre;
             Origin = Anchor.TopCentre;
-
             Masking = true;
-
             EdgeEffect = new EdgeEffectParameters
             {
                 Colour = Color4.Black.Opacity(0),
                 Type = EdgeEffectType.Shadow,
                 Radius = 10
             };
-        }
 
-        [BackgroundDependencyLoader]
-        private void load()
-        {
             Waves.FirstWaveColour = ColourProvider.Light4;
             Waves.SecondWaveColour = ColourProvider.Light3;
             Waves.ThirdWaveColour = ColourProvider.Dark4;
             Waves.FourthWaveColour = ColourProvider.Dark3;
         }
 
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            OngoingOperationTracker.InProgress.BindValueChanged(isInProgress =>
+            {
+                if (isInProgress.NewValue)
+                    loadingLayer.Show();
+                else
+                    loadingLayer.Hide();
+            });
```

Comment:
```suggestion
            }, true);
```
don't see a reason for this not being true.